genshin/models/genshin/chronicle/img_theater.py

```diff
@@ -0,0 +1,145 @@
+import datetime
+import enum
+import typing
+
+if typing.TYPE_CHECKING:
+    import pydantic.v1 as pydantic
+else:
+    try:
+        import pydantic.v1 as pydantic
+    except ImportError:
+        import pydantic
+
+from genshin.constants import CN_TIMEZONE
+from genshin.models.genshin import character
+from genshin.models.model import Aliased, APIModel
+
+__all__ = (
+    "Act",
+    "ActCharacter",
+    "ImgTheater",
+    "TheaterBuff",
+    "TheaterCharaType",
+    "TheaterDifficulty",
+    "TheaterSchedule",
+    "TheaterStats",
+)
+
+
+class TheaterCharaType(enum.IntEnum):
+    """The type of character in the context of the imaginarium theater gamemode."""
+
+    NORMAL = 1
+    TRIAL = 2
+    SUPPORT = 3
+
+
+class TheaterDifficulty(enum.IntEnum):
+    """The difficulty of the imaginarium theater data."""
+
+    EASY = 1
+    NORMAL = 2
+    HARD = 3
+
+
+class ActCharacter(character.BaseCharacter):
+    """A character in an act."""
+
+    type: TheaterCharaType = Aliased("avatar_type")
+    level: int
```

Comment:
BaseCharacter autofills information from the database not the API data.

Reply:
BaseCharacter is partially auto-refilled in `root_validator` with the data passed in, but with a static key `id`, and you can't change that. In this data, the return value is `avatar_id`  instead of  `id`.

Reply:
I am pretty sure it is not like that, the autofill does not use the data passed in but rather from the database, and in the database character IDs use `id` as key.

Reply:
After taking a closer look at the `_get_db_char` function, the fact that the id is missing doesn't affect the result.